xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -722,7 +722,9 @@ public void run() {
           if (stopped || receivedConfig) {
             return;
           }
-          listener.onError(error);
+          listener.onError(Status.UNAVAILABLE.withCause(error.getCause()).withDescription(
+              String.format("Unable to load LDS %s. xDS server returned: %s. %s.",
```

Comment:
nit: don't end in a period. We tend not to, and in this case it could have cause there to be multiple trailing periods if the description included a period.

You might also consider using a colon between the status code and message, as that is what we do when creating an exception. https://github.com/grpc/grpc-java/blob/3b9ff362b9b64b3615c493a4f94eed6d00f711e5/api/src/main/java/io/grpc/Status.java#L431 Not important.

Ditto below.